app-android/src/main/res/layout/fragment_report_detail.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <data>
+
+        <import type="android.view.View" />
+
+        <variable
+            name="report"
+            type="com.ustadmobile.lib.db.entities.ReportWithFilters" />
+
+        <variable
+            name="loading"
+            type="boolean" />
+
+        <variable
+            name="fieldsEnabled"
+            type="boolean" />
+
+    </data>
+
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
There is no need for the constraintlayout here. The recyclerview should be the root view. Here the recyclerview width and height is set to match parent, so the recyclerview will recycle views as intended. When recyclerviews are nested, and height is set to wrap_content, the recycling stops working and this triggers loading everything at once.